src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/object/ObjectDescendantFunctionIterator.java

```diff
@@ -20,7 +20,6 @@ public ObjectDescendantFunctionIterator(List<RuntimeIterator> arguments, Iterato
     public Item next() {
         if (this._hasNext) {
             if (results == null) {
```

Comment:
Likewise -- this initialization should be done at the end of an overriden open() method, as well as setting _hasNext to false if there are no results at all.